src/components/Videos.vue

```diff
@@ -22,52 +22,93 @@
 
 <template>
 	<video
-		v-if="path"
+		v-if="path && canPlay"
 		:autoplay="active"
-		preload
 		:controls="visibleControls"
+		:poster="livePhotoPath"
+		:preload="true"
 		:style="{
 			height: height + 'px',
 			width: width + 'px'
 		}"
+		@canplay="doneLoading"
 		@click.prevent="playPause"
 		@dblclick.prevent="toggleFullScreen"
-		@canplay="doneLoading"
+		@ended="donePlaying"
+		@loadedmetadata="updateVideoSize"
 		@mouseenter="showControls"
-		@mouseleave="hideControls"
-		@loadedmetadata="updateVideoSize">
+		@mouseleave="hideControls">
 
 		<source :src="davPath" :type="mime">
 
 		{{ t('viewer', 'Your browser does not support the video tag.') }}
 	</video>
+
+	<!-- Browser cannot play this file -->
+	<Error v-else>
+		{{ t('viewer', 'This video is not playable in your browser') }}
+	</Error>
 </template>
 
 <script>
-import mime from 'Mixins/Mime'
+import Error from 'Components/Error'
+import Mime from 'Mixins/Mime'
+import PreviewUrl from 'Mixins/PreviewUrl'
+
+const liveExt = ['jpg', 'jpeg', 'png']
 
 export default {
 	name: 'Videos',
-	mixins: [
-		mime
-	],
+
+	components: {
+		Error
+	},
+
+	mixins: [Mime, PreviewUrl],
+
 	data() {
 		return {
+			canPlay: true,
 			visibleControls: false
 		}
 	},
+
+	computed: {
+		livePhoto() {
+			return this.fileList.find(file => {
+				// if same filename and extension is allowed
+				return file.name.startsWith(this.name)
+					&& liveExt.indexOf(file.name.split('.')[1] > -1)
```

Comment:
won't this :boom: with files without an extension?

Reply:
No it will just output undefined, so `undefined > -1`
```js
$ undefined > -1
false
```